### PR TITLE
add BigIntegerConverter

### PR DESCRIPTION
**Pull request by arnzel**
Add Converter for BigInteger

### CODE REVIEW

**Comment by zapodot** [comment]
@arnzel Thank you for your contribution. As we try to improve our test coverage we require all PR-s to include tests. Can you please add some and update this PR?

**Comment by arnzel** [comment]
yes sure

**Comment by arnzel** [comment]
@zapodot : i added some tests and library assertJ for better failing errors

**Comment by zapodot** [comment]
@arnzel I see from your commit that you have added a dependency on AssertJ. Any chance you could use the basic JUnit assertions instead? Thanks again for you contribution 👍

File: core/src/test/java/org/sql2o/converters/BigIntegerConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigInteger;
+
+public class BigIntegerConverterTest {
```
Comment (zapodot) [inline core/src/test/java/org/sql2o/converters/BigIntegerConverterTest.java]: Could you also add a test where you try to convert a null value?

**Comment by arnzel** [comment]
> @arnzel I see from your commit that you have added a dependency on AssertJ. Any chance you could use the basic JUnit assertions instead? Thanks again for you contribution

well i can do it,but i dont like JUNIt Assertions like "assertEquals" and "assertTrue". they only tell you somethig like "expected true but was false".AssertJ gives you much more information and is better maintained like other matching libraries like hamcrest

**Comment by zapodot** [comment]
There are an assertThat assertion in JUnit as well if you prefer that :-)

**Comment by arnzel** [comment]
@zapodot  okay i removed assertJ and use hamcrest matchers. also i added converting null values

**Comment by arnzel** [comment]
@zapodot  : i dont understand travic ci test error, is it related to my change ?

**Comment by zapodot** [comment]
@arnzel we are experiencing some weird behaviour in our builds right now. Will await merge before until we have a green master branch. Good job!

File: core/src/main/java/org/sql2o/converters/BigIntegerConverter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @Override
+    protected BigInteger convertNumberValue(Number number) {
+        if(null == number){
+            return null;
+        }
+        else if (number instanceof BigInteger){
+            return (BigInteger)number;
+        }
+        else{
+            return BigInteger.valueOf(number.intValue());
```
Comment (Gr1f0n6x) [inline core/src/main/java/org/sql2o/converters/BigIntegerConverter.java]: Is seems that `number.longValue()` would be better than  `number.intValue()`. Otherwise you can lost precision.
Also, it's more native to use long than int when you want to get BigInteger

File: core/src/main/java/org/sql2o/converters/BigIntegerConverter.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+    }
+
+    @Override
+    protected BigInteger convertStringValue(String string) {
+        if(null != string) {
+            return BigInteger.valueOf(Integer.parseInt(string));
```
Comment (Gr1f0n6x) [inline core/src/main/java/org/sql2o/converters/BigIntegerConverter.java]: Same here. `Long.parseLong(...)` would be better

**Comment by MarcoSulla** [comment]
Is this not already fixed?
